providers/default/digests/blake2_prov.c

```diff
@@ -7,10 +7,15 @@
  * https://www.openssl.org/source/license.html
  */
 
-#include <openssl/crypto.h>
-#include "internal/blake2.h"
-#include "internal/core_mkdigest.h"
-#include "internal/provider_algs.h"
+#include <openssl/opensslconf.h>
+
+#ifdef OPENSSL_NO_BLAKE2
+NON_EMPTY_TRANSLATION_UNIT
+#else
```

Comment:
Why?  If you look in `providers/default/digests/build.info`, you'll see that disabled algos aren't built (if they are, there's a bug in build.info processing)

This goes for all places where you added this.

Reply:
hangs head in shame - and goes and reverts changes...